From 19414af15a7764ba62686f504a0ba0a39dcef2b5 Mon Sep 17 00:00:00 2001
From: Christophe Kerello <christophe.kerello@foss.st.com>
Date: Thu, 21 Nov 2024 11:08:46 +0100
Subject: [PATCH 1297/1695] plat-stm32mp2: stm32mp25_pwr: restore fixed
 regulator

Only IOD regulators are currently restored on standby mode.
Fixed regulators should also be restored on standby mode.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git fe839bfef3ec82d4cd9d1e35ff88de10c8c6d69b plat-stm32mp2: stm32mp25_pwr: restore fixed regulator]
Fixes: d4bb9f98f4af ("plat-stm32mp2: Add PWR regulators driver")
Change-Id: Ia016e50e9e1e47e954c52671168f0933bdf9e608
Signed-off-by: Christophe Kerello <christophe.kerello@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/420013
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/423126
Tested-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/arch/arm/plat-stm32mp2/drivers/stm32mp25_pwr_regul.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/core/arch/arm/plat-stm32mp2/drivers/stm32mp25_pwr_regul.c b/core/arch/arm/plat-stm32mp2/drivers/stm32mp25_pwr_regul.c
index 3b2214c04..a6ffcd493 100644
--- a/core/arch/arm/plat-stm32mp2/drivers/stm32mp25_pwr_regul.c
+++ b/core/arch/arm/plat-stm32mp2/drivers/stm32mp25_pwr_regul.c
@@ -418,6 +418,7 @@ DECLARE_KEEP_PAGER(pwr_regu_ops);
 static const struct regulator_ops pwr_regu_fixed_ops = {
 	.set_state = pwr_set_state,
 	.get_state = pwr_get_state,
+	.supplied_init = pwr_supplied_init,
 };
 DECLARE_KEEP_PAGER(pwr_regu_fixed_ops);
 
-- 
2.47.2

